tests/testSphPoint.py

```diff
@@ -0,0 +1,615 @@
+            dict(lon=lon0, lat=lat0, bearing=45.0, lonEnd=lon0+180.0, latEnd=-lat0),
+            #
+            dict(lon=lon0, lat=lat0, bearing=45.0, lonEnd=lon0+90.0, latEnd=lat0+45.0),
+            dict(lon=lon0, lat=lat0, bearing=225.0, lonEnd=lon0-90.0, latEnd=lat0-45.0),
+        ]
+
+        for trial in trials:
+            origin = SphPoint(trial['lon']*degrees, trial['lat']*degrees)
+            end = SphPoint(trial['lonEnd']*degrees, trial['latEnd']*degrees)
+            bearing = origin.bearingTo(end)
+
+            self.assertIsInstance(bearing, afwGeom.Angle)
+            self.assertGreaterEqual(bearing.asDegrees(), 0.0)
+            self.assertLess(bearing.asDegrees(), 360.0)
+            if origin.separationTo(end).asDegrees() != 180.0:
+                self.assertAlmostEqual(trial['bearing'], bearing.asDegrees(), 12)
+
+    def testBearingToValueSingular(self):
+        """White-box test: bearingTo() may be unstable if points are near opposite poles.
+        """
+        southPole = SphPoint(0.0*degrees, (-90.0 + 1e-14)*degrees)
+        northPoleSame = SphPoint(0.0*degrees, (90.0 - 1e-14)*degrees)
+        # Don't let it be on exactly the opposite side
+        northPoleOpposite = SphPoint(180.0*degrees, (90.0 - 1e-13)*degrees)
+
+        self.assertAnglesNearlyEqual(southPole.bearingTo(northPoleSame), afwGeom.HALFPI*afwGeom.radians)
+        self.assertAnglesNearlyEqual(southPole.bearingTo(northPoleOpposite),
+                                     (afwGeom.PI + afwGeom.HALFPI)*afwGeom.radians)
+
+    def testSeparationToValueGeneric(self):
+        """Test if separationTo() returns the correct value.
+        """
+        # This should cover arcs over the meridian, across the pole, etc.
+        # Do not use sphgeom as an oracle, in case SphPoint uses it
+        # internally.
+        for lon1, lat1 in self._dataset[0:20]:
+            point1 = SphPoint(lon1*degrees, lat1*degrees)
+            x1, y1, z1 = SphPointTestSuite._toVector(lon1, lat1)
+            for lon2, lat2 in self._dataset[0:20]:
+                point2 = SphPoint(lon2*degrees, lat2*degrees)
+                if lon1 != lon2 or lat1 != lat2:
+                    # Numerically unstable at small angles, but that's ok.
+                    x2, y2, z2 = SphPointTestSuite._toVector(lon2, lat2)
+                    expected = math.acos(x1*x2 + y1*y2 + z1*z2)
+                else:
+                    expected = 0.0
+
+                sep = point1.separationTo(point2)
+                self.assertIsInstance(sep, afwGeom.Angle)
+                self.assertGreaterEqual(sep.asDegrees(), 0.0)
+                self.assertLessEqual(sep.asDegrees(), 180.0)
+
+                self.assertAlmostEqual(expected, sep.asRadians())
+                self.assertAnglesNearlyEqual(sep, point2.separationTo(point1))
+
+    def testSeparationToValueAbsolute(self):
+        """Test if separationTo() returns specific values.
+        """
+        # Test from Meeus, p. 110
```

Comment:
@r-owen This test was copied from `Coord`. I kept it mainly because it provides a human-validated oracle, but maybe it's not worth it?